eureka-client-archaius2/src/main/java/com/netflix/appinfo/EurekaArchaius2InstanceConfig.java

```diff
@@ -38,88 +39,90 @@ public Name getName() {
     }
     
     public EurekaArchaius2InstanceConfig(Config config, String namespace, DataCenterInfo dcInfo) {
-        this.defaultAppGroup = config.getString("NETFLIX_APP_GROUP", null);
+        this.defaultAppGroup = config.getString(FALLBACK_APP_GROUP_KEY, Values.UNKNOWN_APPLICATION);
         
         this.namespace = namespace;
         this.config = config.getPrefixedView(namespace);
         this.dcInfo = dcInfo;
-        
-        // TODO: archaius.deployment.environment=${env}?
+
+        // TODO: Remove this when DiscoveryManager is finally no longer used
+        DiscoveryManager.getInstance().setEurekaInstanceConfig(this);
```

Comment:
If we need to do this, I feel like we can at least do this somewhere else. A binding for something like DiscoveryManagerInitializer with dependencies on these Config classes might work.